com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java

```diff
@@ -178,6 +188,37 @@ public void clearState(ThreadReference thread) {
         }
     }
 
+    private boolean ensureDebugTarget(VirtualMachine vm) {
+        String projectName = (String) options.get(Constants.PROJECTNAME);
+        if (debugTarget == null) {
+            if (project == null) {
+                if (StringUtils.isBlank(projectName)) {
+                    logger.severe("Cannot evaluate when project is not specified.");
+                    throw new IllegalStateException("Please specify projectName in launch.json.");
+                }
+                project = JdtUtils.getJavaProject(projectName);
+            }
+
+            if (project == null) {
+                new IllegalStateException(String.format("Project %s cannot be found.", projectName));
```

Comment:
Why new a unused exception?

Reply:
will use exception for the failure of initialization to  debugTarget